pkg/services/rpcsrv/server.go

```diff
@@ -2505,22 +2513,33 @@ func (s *Server) postProcessExecStack(stack []stackitem.Item) *session {
 
 // registerOrDumpIterator changes iterator interop stack items into result.Iterator
 // interop stack items and returns a uuid for it if sessions are enabled. All the other stack
-// items are not changed.
-func (s *Server) registerOrDumpIterator(item stackitem.Item) (stackitem.Item, uuid.UUID) {
+// items are not changed. The third return value is the current iterator value if it's not nil.
+func (s *Server) registerOrDumpIterator(item stackitem.Item) (stackitem.Item, uuid.UUID, stackitem.Item) {
 	var iterID uuid.UUID
 
 	if (item.Type() != stackitem.InteropT) || !iterator.IsIterator(item) {
-		return item, iterID
+		return item, iterID, nil
 	}
-	var resIterator result.Iterator
+	var (
+		resIterator result.Iterator
+		curr        stackitem.Item
+	)
 
-	if s.config.SessionEnabled {
-		iterID = uuid.New()
-		resIterator.ID = &iterID
+	if s.config.SessionExpansionEnabled {
```

Comment:
Add verification to the config level that `SessionExpansionEnabled` may be enabled only if `SessionEnabled` is enabled.